internal/cli/e2e.go

```diff
@@ -0,0 +1,262 @@
+	fs.BoolVarP(&o.destroyEnvironment, "destroy-env", "d", defaultDestroyEnv, fmt.Sprintf("Destroy the created %s cluster", o.clusterName))
+}
+
+func (o *E2ETestOptions) Run(ctx context.Context, args []string) error {
+	err := o.configureEnvironment()
+	if err != nil {
+		return err
+	}
+
+	o.deployEnvironment(ctx)
+
+	err = o.waitForMigrationPlannerService(ctx)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < o.testCount; i++ {
+		o.executeTest(ctx)
+	}
+
+	if o.destroyEnvironment {
+		destroyEnvironment(ctx)
+	}
+
+	return nil
+}
+
+func getPackageManager() string {
+	if _, err := exec.LookPath("dnf"); err == nil {
+		return "dnf"
+	} else if _, err := exec.LookPath("apt"); err == nil {
+		return "apt"
+	}
+	return ""
+}
+
+func runCommand(ctx context.Context, cmdStr string) error {
+	cmd := exec.CommandContext(ctx, "bash", "-c", cmdStr)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func (o *E2ETestOptions) configureEnvironment() error {
+	envVars := map[string]string{
+		"REGISTRY_IP":                             o.registryIP.String(),
+		"INSECURE_REGISTRY":                       o.insecureRegistryAddr,
+		"MIGRATION_PLANNER_AGENT_IMAGE":           o.agentImage,
+		"MIGRATION_PLANNER_API_IMAGE":             o.plannerAPIImage,
+		"MIGRATION_PLANNER_API_IMAGE_PULL_POLICY": o.plannerAPIImagePullPolicy,
+		"PODMAN":      o.containerRuntime,
+		"PKG_MANAGER": o.pkgManager,
+		"IFACE":       o.iface,
+	}
+
+	for key, value := range envVars {
+		if err := os.Setenv(key, value); err != nil {
+			return fmt.Errorf("failed to set env variable %s: %w", key, err)
+		}
+	}
+
+	o.printConfig(envVars)
+
+	return nil
+}
+
+func (o *E2ETestOptions) printConfig(envVars map[string]string) {
+	// Print the Environment Variables Configuration
+	fmt.Printf("====================================\n")
+	fmt.Printf("🔧 Environment Variables Configured:\n")
+	fmt.Printf("====================================\n")
+	for key, value := range envVars {
+		fmt.Printf("%s=%s\n", key, value)
+	}
+	fmt.Printf("====================================\n")
+
+	// Print the Test Execution Configuration
+	fmt.Printf("====================================\n")
+	fmt.Printf("🛠 Test Execution Configuration:\n")
+	fmt.Printf("====================================\n")
+	fmt.Printf("Number of tests to run: %d\n", o.testCount)
+	fmt.Printf("Destroy environment at the end? %v\n", o.destroyEnvironment)
+	fmt.Printf("====================================\n")
+}
+
+func kindClusterExists(ctx context.Context, clusterName string) bool {
+	cmd := exec.CommandContext(ctx, "kind", "get", "clusters")
+	output, err := cmd.Output()
+	if err != nil {
+		log.Printf("Error checking Kind clusters: %v\n", err)
+		return false
+	}
+
+	return strings.Contains(string(output), clusterName)
+}
+
+func (o *E2ETestOptions) deployEnvironment(ctx context.Context) {
+	if !(kindClusterExists(ctx, o.clusterName)) {
```

Comment:
please avoid negation when you can use the affirmative.
``` 
if kindClusterExists(..) {
   return
```
it's way simpler

Reply:
The condition can be put outside the method. 
The signature gives the impression that you're deploying the env, but you don't depending of the existence of the kind cluster.
Also, this method will be a bit hard to test because of that condition.